src/producer.rs

```diff
@@ -254,20 +257,19 @@ impl<P: Partitioner> Producer<P> {
               V: AsBytes
     {
         let mut rs = try!(self.send_all(ref_slice(rec)));
-        if self.config.required_acks == 0 {
+
+        if self.config.required_acks != 0 {
             // ~ with no required_acks we get no response and
             // consider the send-data request blindly as successful
-            Ok(())
-        } else {
             assert_eq!(1, rs.len());
-            rs.pop().unwrap().offset.map(|_| ())
+            rs.pop().unwrap();
         }
+
+        Ok(())
```

Comment:
Actually, I hadn't considered the case where acknowledgement isn't required. I'll just leave the return type as is for now and fix it as you suggested.